suite-native/module-send/src/hooks/useCustomFee.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useCallback, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { isRejected } from '@reduxjs/toolkit';
+import { G } from '@mobily/ts-belt';
+
+import { AccountsRootState, selectAccountNetworkType } from '@suite-common/wallet-core';
+import {
+    AccountKey,
+    isFinalPrecomposedTransaction,
+    TokenAddress,
+} from '@suite-common/wallet-types';
+import { useTranslate } from '@suite-native/intl';
+import { useDebounce } from '@trezor/react-utils';
+import { BigNumber } from '@trezor/utils';
+import { useFormContext } from '@suite-native/forms';
+
+import { SendFeesFormValues } from '../sendFeesFormSchema';
+import {
+    selectCustomFeeLevel,
+    NativeSendRootState,
+    selectFeeLevelTransactionBytes,
+} from '../sendFormSlice';
+import { calculateCustomFeeLevelThunk } from '../sendFormThunks';
+
+type UseCustomFeeProps = {
+    accountKey: AccountKey;
+    tokenContract?: TokenAddress;
+};
+
+export const useCustomFee = ({ accountKey, tokenContract }: UseCustomFeeProps) => {
+    const dispatch = useDispatch();
+    const debounce = useDebounce();
+    const { translate } = useTranslate();
+
+    const [isErrorBoxVisible, setIsErrorBoxVisible] = useState(false);
+    const [isFeeLoading, setIsFeeLoading] = useState(false);
+
+    const customFeeLevel = useSelector(selectCustomFeeLevel);
+    const networkType = useSelector((state: AccountsRootState) =>
+        selectAccountNetworkType(state, accountKey),
+    );
+
+    const {
+        formState: { errors },
+        setError,
+        trigger,
+        getValues,
+        watch,
+    } = useFormContext<SendFeesFormValues>();
+
+    const feePerUnitFieldName = 'customFeePerUnit';
+    const feeLimitFieldName = 'customFeeLimit';
+
+    const watchedFeePerUnit = watch(feePerUnitFieldName, '0');
+    const watchedFeeLimit = watch(feeLimitFieldName, '1') as string;
```

Comment:
Because the feeLimit is optional so, its type is equal to `string | undefined`. For this reason, there is a default value `1` passed to the watch function, so the result of the function should always be a `string`. `react-hook-form` TypeScript implementation is not the best, so it returns the type of `string | undefined` anyway. The typecast fixes that.